test/conftest.py

```diff
@@ -239,20 +264,24 @@ def pudl_sql_engine(
     logger.info("setting up the pudl_engine fixture")
     if not live_dbs:
         # Run the ETL and generate a new PUDL SQLite DB for testing:
-        pudl.etl.etl(
-            etl_settings=etl_settings,
-            pudl_settings=pudl_settings_fixture,
-            clobber=False,
-            check_foreign_keys=check_foreign_keys,
-            check_types=True,
-            check_values=True,
-            use_local_cache=not request.config.getoption("--bypass-local-cache"),
-            gcs_cache_path=request.config.getoption("--gcs-cache-path"),
+        execute_job(
+            reconstructable(get_etl_job),
+            instance=DagsterInstance.get(),
+            run_config={
+                "resources": {
+                    "dataset_settings": {
+                        "config": dataset_settings_config,
+                    },
+                    "datastore": {
+                        "config": pudl_datastore_config,
+                    },
+                }
+            },
         )
     # Grab a connection to the freshly populated PUDL DB, and hand it off.
     # All the hard work here is being done by the datapkg and
     # datapkg_to_sqlite fixtures, above.
-    engine = sa.create_engine(pudl_settings_fixture["pudl_db"])
+    engine = sa.create_engine(f"sqlite:///{os.getenv('PUDL_OUTPUT')}/pudl.sqlite")
```

Comment:
We could get the engine from the `pudl_sqlite_io_manager`.

Reply:
Also, if we have the `pudl_sqlite_io_manager` we could run `pudl_sqlite_io_manager.check_foreign_keys()` once the ETL is finished so we know the FKs are working. However, it might make sense for check the foreign keys in a separate test so it's clear what is failing.

Reply:
Good call, I added this check in `test_pudl_engine`